packages/kbn-config-schema/src/types/array_type.test.ts

```diff
@@ -24,29 +24,65 @@ test('returns value if it matches the type', () => {
   expect(type.validate(['foo', 'bar', 'baz'])).toEqual(['foo', 'bar', 'baz']);
 });
 
+test('properly parse the value if input is a string', () => {
+  const type = schema.arrayOf(schema.string());
+  expect(type.validate('["foo", "bar", "baz"]')).toEqual(['foo', 'bar', 'baz']);
+});
+
 test('fails if wrong input type', () => {
   const type = schema.arrayOf(schema.string());
-  expect(() => type.validate('test')).toThrowErrorMatchingSnapshot();
+  expect(() => type.validate(12)).toThrowErrorMatchingInlineSnapshot(
+    `"expected value of type [array] but got [number]"`
+  );
+});
+
+test('fails if string input cannot be parsed', () => {
+  const type = schema.arrayOf(schema.string());
+  expect(() => type.validate('test')).toThrowErrorMatchingInlineSnapshot(
+    `"could not parse array value from [test]"`
+  );
+});
+
+test('fails with correct type if parsed input is not an array', () => {
+  const type = schema.arrayOf(schema.string());
+  expect(() => type.validate('{"foo": "bar"}')).toThrowErrorMatchingInlineSnapshot(
+    `"expected value of type [array] but got [Object]"`
+  );
 });
 
 test('includes namespace in failure when wrong top-level type', () => {
   const type = schema.arrayOf(schema.string());
-  expect(() => type.validate('test', {}, 'foo-namespace')).toThrowErrorMatchingSnapshot();
+  expect(() => type.validate('test', {}, 'foo-namespace')).toThrowErrorMatchingInlineSnapshot(
+    `"[foo-namespace]: could not parse array value from [test]"`
+  );
```

Comment:
I switched to inline snapshots. Should probably have done this in a separate commit though. No snapshot did change except the ones that were using string data as invalid type, as the error changed to `could not parse...`. I did adapt the data on these tests though.